ansible/files/bootstrap/telemetry.sh

```diff
@@ -209,6 +221,35 @@ rm RRO-3.2.1-el6.x86_64.tar.gz
 cd RRO-3.2.1; sudo ./install.sh; cd $HOME
 $ANACONDA_PATH/bin/pip install rpy2
 
+# Setup metrics reporting
+if [ "$METRICS_PROVIDER" = "datadog" ] ; then
+    # Pull down and decrypt Datadog API key; see https://github.com/mozilla/emr-bootstrap-spark/pull/485
+    DD_DIR=$(mktemp -d)
+    aws s3 cp $TELEMETRY_CONF_BUCKET/packages/sops-3.0.4-1.x86_64.rpm $DD_DIR/
+    aws s3 cp $TELEMETRY_CONF_BUCKET/credentials/emr.yaml $DD_DIR/
+    sudo yum -y localinstall $DD_DIR/sops-3.0.4-1.x86_64.rpm
+    export DD_API_KEY=$(/usr/local/bin/sops --decrypt --extract '["datadog_agent::api_key"]' $DD_DIR/emr.yaml)
+    # Install datadog-agent; see https://app.datadoghq.com/account/settings#agent/aws
+    bash -c "$(curl -sLS https://raw.githubusercontent.com/DataDog/datadog-agent/master/cmd/agent/install_script.sh)"
+    rm -r "$DD_DIR"
+
+    # Create Datadog spark config; see https://docs.datadoghq.com/integrations/spark/
+    cat > spark.yaml << EOL
+init_config:
+instances:
+- spark_url: http://$(hostname):8088
+cluster_name: ${CLUSTER_NAME}
+spark_cluster_mode: spark_yarn_mode
+tags:
+$(describe_cluster '.Cluster.Tags[] | "- \(.Key):\(.Value)"')
+- cluster_id:${CLUSTER_ID}
+EOL
```

Comment:
For an example cluster I spun up similar to an ATMO cluster, this heredoc expanded to:

```yaml
init_config:
instances:
  - spark_url: http://ip-172-31-33-18:8088
    cluster_name: klukas-metrics-test
spark_cluster_mode: spark_yarn_mode
tags:
- App:telemetry-analysis
- Environment:prod
- Owner:jklukas@mozilla.com
- Type:worker
- Application:telemetry-analysis-worker-instance
- Name:klukas-metrics-test
- cluster_id:j-330MAU776X1LH
```